### PR TITLE
Add dotnetcore-sdk recipe.

### PR DESCRIPTION
**Pull request by cgranade**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

This PR adds a new package, `dotnetcore-sdk`, which provides .NET Core SDK as the `dotnet` command in `$PREFIX/opt/dotnet`.
In lieu of conda/conda#6820, this PR also includes activation and deactivation scripts for Bash, cmd, and PowerShell that set `PATH` to include `dotnet`, and that add the `DOTNET_ROOT` environment variable required by the .NET Core SDK.

@conda-forge-admin, please ping team

### CODE REVIEW

**Comment by cgranade** [comment]
If it'd be possible, I'd greatly appreciate some advice as to why the overlinking check is failing, as this recipe builds fine on my local machine. Thanks!

**Comment by scopatz** [comment]
> I'd greatly appreciate some advice as to why the overlinking check is failing, as this recipe builds fine on my local machine

Normally when you build locally, the overlinking check is a warning. On conda-forge we have it as an error, which is probably why it builds fine for you

**Comment by scopatz** [comment]
Try adding `{{ compiler('c') }}` to the build requirements to get the approiate linkages

File: recipes/dotnetcore-sdk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  home: https://github.com/dotnet/core-sdk
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: '.NET Core is a cross-platform version of .NET for building websites, services, and console apps.'
+
+  description: |
+    .NET Core is a free and open-source, managed computer software framework for Windows, Linux, and macOS operating systems.
+  doc_url: https://simplejson.readthedocs.io/
+  dev_url: https://github.com/simplejson/simplejson
```
Comment (isuruf) [inline recipes/dotnetcore-sdk/meta.yaml]: These two URLs are wrong

Comment (cgranade) [reply recipes/dotnetcore-sdk/meta.yaml]: Absolutely, thanks for catching that! Copied and pasted for the tutorial and forgot to change those two.

**Comment by cgranade** [comment]
> What's the oldest MacOS version that these repackaged binaries support?

The most up-to-date listing is at https://docs.microsoft.com/dotnet/core/install/dependencies, which lists that 3.1.100 requires High Sierra (10.13+). Is there somewhere I can list that in the install requirements?

File: recipes/dotnetcore-sdk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  # allow for cross-compiling and multi-targeting.
+  # Here, we whitelist the few native dependencies brought in by these packages.
+  - "*/WINTRUST.dll" # used by comhost.dll, a native component in Microsoft.NETCore.App.Host.win-x64.
+  - "*/VERSION.dll" # used by dbgshim.dll, a native component used to provide native debugging shims.
+  - "*/SHLWAPI.dll" # part of the core Windows system libraries, used by coreclr.dll
+
+
+requirements:
+  build:
+  - {{ compiler('c') }}
```
Comment (isuruf) [inline recipes/dotnetcore-sdk/meta.yaml]:
```suggestion
  - {{ compiler('c') }}
  run_constrained:
  - __osx >=10.13  # [osx]
```

Comment (cgranade) [reply recipes/dotnetcore-sdk/meta.yaml]: Thank you!

**Comment by cgranade** [comment]
Thanks for the help, @scopatz and @isuruf! I think I've gotten this to where there's a few system dependencies (namely those listed in https://docs.microsoft.com/dotnet/core/install/dependencies) are missing, causing overlinkage errors. I tried adding libcurl and others to the recipe, but that doesn't seem to have resolved the problem. I'll keep working on it, but in the meantime, thank you again for your help!

**Pull request closed without merging**